code/modules/clothing/spacesuits/plasmamen.dm

```diff
@@ -23,21 +22,46 @@
 	. = ..()
 	. += span_notice("There [extinguishes_left == 1 ? "is" : "are"] [extinguishes_left] extinguisher charge\s left in this suit.")
 
+/obj/item/clothing/suit/space/eva/plasmaman/equipped(mob/living/user, slot)
+	. = ..()
+	if (slot & ITEM_SLOT_OCLOTHING)
+		START_PROCESSING(SSprocessing, src)
+
+/obj/item/clothing/suit/space/eva/plasmaman/dropped(mob/living/user)
+	. = ..()
+	STOP_PROCESSING(SSprocessing, src)
+
+/obj/item/clothing/suit/space/eva/plasmaman/process(seconds_per_tick)
```

Comment:
Wouldn't it be better to put being on fire detection on a signal instead, that listens on when the owner is lit on fire? That would let you avoid processing the suit at all. If it's missing a signal it should be easy to add one

Reply:
definitely, had a bit of a brainfreeze there - thanks

Reply:
right, this was a problem because of cooldowns and our timer-based cooldowns are really not great